### PR TITLE
[3.11] Bug 1690066: Report build failures due to eviction

### PR DESCRIPTION
**Pull request by adambkaplan**
Add new reason to indicate a build fails because its pod was evicted.

/assign @smarterclayton 
/cc @bparees 
/cc @sjenning

### CODE REVIEW

**Comment by smarterclayton** [comment]
What is the "DELETE ME" commit?

**Comment by bparees** [comment]
@smarterclayton that is Adam's temporary commit that's doing a bump from his own branch of openshift/api, presumably because his dependent changes haven't merged there.  Once openshift/api is updated, he'll delete that commit and do a proper bump from the real openshift/api 3.11 branch.

**Comment by adambkaplan** [comment]
@smarterclayton updated with real bump.

File: vendor/github.com/vmware/photon-controller-go-sdk/photon/client.go [hunk 1]
```diff
@@ -142,8 +142,8 @@ func NewClient(endpoint string, options *ClientOptions, logger *log.Logger) (c *
 	}
 
 	restClient := &restClient{
-		httpClient:                &http.Client{Transport: tr},
-		logger:                    logger,
+		httpClient: &http.Client{Transport: tr},
+		logger:     logger,
```
Comment (adambkaplan) [inline vendor/github.com/vmware/photon-controller-go-sdk/photon/client.go]: @stevekuznetsov @deads2k These go-fmt changes in github.com/vmware/photon were previously added when they were not supposed to.

**Comment by adambkaplan** [comment]
ping @smarterclayton @stevekuznetsov

**Comment by adambkaplan** [comment]
/hold

Bug https://bugzilla.redhat.com/show_bug.cgi?id=1693823

**Pull request merged**